modules/contracts/src.sol/transferDefinitions/HashlockTransfer.sol

```diff
@@ -47,15 +47,15 @@ contract HashlockTransfer is TransferDefinition {
     if (resolver.preImage != bytes32(0) && (state.expiry == 0 || state.expiry > block.number)) {
       // Check hash for normal payment unlock
       bytes32 generatedHash = sha256(abi.encode(resolver.preImage));
-      require(state.lockHash == generatedHash, "Hash generated from preimage does not match hash in state");
+      require(state.lockHash == generatedHash, "HashlockTransfer: INVALID_PREIMAGE");
 
       // Update state
       balance.amount[1] = balance.amount[0];
       balance.amount[0] = 0;
     } else {
       // To cancel, the preImage must be empty (not simply incorrect)
-      require(resolver.preImage == bytes32(0), "Must provide empty hash to cancel payment");
-
+      // TODO: rm this assertion bc it will always succeed given the if statement that leads us here
+      require(resolver.preImage == bytes32(0), "HashlockTransfer: NONZERO_LOCKHASH");
```

Comment:
We could get here because it has expired and someone is trying to resolve as well